shell.nix

```diff
@@ -0,0 +1,10 @@
+{ pkgs ? import <nixpkgs> {} }:
+with pkgs;
+mkShell {
+  buildInputs = [
+    gcc
+    go_1_18
+    gopls
+    gotests
```

Comment:
would this need the linting tool for golangci-lint and goimports for the make checks?

Reply:
Good point let me check